test/Confluent.Kafka.IntegrationTests/Tests/AdminClient_ListDescribeConsumerGroups.cs

```diff
@@ -0,0 +1,178 @@
+                Assert.Empty(groups.Valid.Where(group => group.GroupId == nonExistentGroupID));
+                var group = groups.Valid.Find(group => group.GroupId == groupID);
+                Assert.Equal(ConsumerGroupState.Stable, group.State);
+                Assert.False(group.IsSimpleConsumerGroup);
+
+                Console.Error.WriteLine("Here 0");
+                var descResult = adminClient.DescribeConsumerGroupsAsync(
+                    new List<String>() { groupID },
+                    describeOptionsWithTimeout).Result;
+                var groupDesc = descResult.ConsumerGroupDescriptions.Find(group => group.GroupId == groupID);
+                var clientIdToToppars = new Dictionary<string, List<TopicPartition>>();
+                clientIdToToppars[clientID1] = new List<TopicPartition>() {
+                    new TopicPartition(partitionedTopic, 0),
+                    new TopicPartition(partitionedTopic, 1),
+                };
+                checkConsumerGroupDescription(
+                    groupDesc, ConsumerGroupState.Stable, "range", groupID, clientIdToToppars);
+
+                Console.Error.WriteLine("Here 1");
+                // 2. One consumer group with two clients.
+                consumerConfig.ClientId = clientID2;
+                var consumer2 = new ConsumerBuilder<byte[], byte[]>(consumerConfig).Build();
+                consumer2.Subscribe(new string[] { partitionedTopic });
+
+                // Wait for rebalance.
+                var state = ConsumerGroupState.PreparingRebalance;
+                while (state != ConsumerGroupState.Stable)
+                {
+                    consumer1.Consume(TimeSpan.FromSeconds(1));
+                    consumer2.Consume(TimeSpan.FromSeconds(1));
+
+                    descResult = adminClient.DescribeConsumerGroupsAsync(
+                        new List<String>() { groupID },
+                        describeOptionsWithTimeout).Result;
+                    Assert.Single(descResult.ConsumerGroupDescriptions.Where(group => group.GroupId == groupID));
+                    groupDesc = descResult.ConsumerGroupDescriptions.Find(group => group.GroupId == groupID);
+                    state = groupDesc.State;
+                }
+
+                clientIdToToppars[clientID1] = new List<TopicPartition>() {
+                    new TopicPartition(partitionedTopic, 0)
+                };
+                clientIdToToppars[clientID2] = new List<TopicPartition>() {
+                    new TopicPartition(partitionedTopic, 1)
+                };
+                checkConsumerGroupDescription(
+                    groupDesc, ConsumerGroupState.Stable, "range", groupID, clientIdToToppars);
+
+                // 3. Empty consumer group.
+                consumer1.Close();
+                consumer2.Close();
+                consumer1.Dispose();
+                consumer2.Dispose();
+
+
+                // TODO(milind) do we need a wait here for rebalancing again?
```

Comment:
Remove (milind). What is the wait here? Maybe remove the TODO too

Reply:
I removed the TODO (in the new PR).

But the question here is, after Closing and Disposing the consumers, there will be a change in the state of the consumer group, from Stable -> Empty (maybe there will be a Rebalancing state in the middle)
Now, I am not sure how much time it will take after the Close() call to trigger a rebalance
Is it possible that, a few milliseconds after the consumers Close(), the group on the broker is still in the Stable state, not having received the message for the consumers' close?
That will cause intermittent test failures.

Reply:
Not sure how long does it take, it this happens we can retry multiple times with a small interval in between before failing the assert